src/zhinst/toolkit/nodetree/helper.py

```diff
@@ -178,3 +178,18 @@ def to_dict(self) -> t.Dict[str, t.Any]:
         After conversion, :class:`Node` objects cannot be used to get items.
         """
         return self._result
+
+
+def not_callable_in_transactions(func: t.Callable[['Node', ...], None]):
+    """Wrapper to prevent certain functions from being used within a transaction.
+
+    Certain utils functions which that both get and set values would not work like
+    expected in a transaction. This wrapper prevents misuse by throwing an error
+    in such cases.
+    """
+    def f(self: 'Node', *args, **kwargs):
+        if self.root.transaction.in_progress():
```

Comment:
this raises an (probably not very descriptive) Attribute error if the decorator is added to a function that is not a member of a `Node`. I would catch that and create a custom more helpful error message for the attribute error

Reply:
we are the only ones using this wrapper and we will only apply it to functions within a class. The dot notation will ensure that the first argument is actually a node ( node1.attr1(*other_arguments) )

Reply:
Why should be be the only ones using this decorator? And we might use the decorator also on other things than a node class. E.g. a function that takes a node as a first argument.

It can also be a useful thing for higher-level APIs 😇 . 
And a try-catch block adds little to no overhead.